x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/date/DateTruncTests.java

```diff
@@ -22,13 +26,52 @@
 import java.time.Instant;
 import java.time.Period;
 import java.util.Collections;
+import java.util.List;
 import java.util.Map;
+import java.util.function.Supplier;
 
 import static org.elasticsearch.xpack.esql.expression.function.scalar.date.DateTrunc.createRounding;
 import static org.elasticsearch.xpack.esql.expression.function.scalar.date.DateTrunc.process;
 import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+
+public class DateTruncTests extends AbstractFunctionTestCase {
 
-public class DateTruncTests extends ESTestCase {
+    public DateTruncTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testCaseSupplier) {
+        this.testCase = testCaseSupplier.get();
+    }
+
+    @ParametersFactory
+    public static Iterable<Object[]> parameters() {
+        return parameterSuppliersFromTypedData(
+            List.of(
+                new TestCaseSupplier(
+                    List.of(EsqlDataTypes.DATE_PERIOD, DataTypes.DATETIME),
+                    () -> new TestCaseSupplier.TestCase(
+                        List.of(
+                            new TestCaseSupplier.TypedData(Period.ofYears(1), EsqlDataTypes.DATE_PERIOD, "interval"),
```

Comment:
Maybe it wouldn't hurt to randomize the date period?

Applies to the time duration test case similarly.

Reply:
The original tests are pretty complete (better than randomized), I think I'll just have to refactor them into test case suppliers.